backend/private-graph/graph/resolver.go

```diff
@@ -3667,3 +3728,141 @@ func (r *Resolver) GetJiraProjects(
 
 	return jira.GetJiraProjects(workspace, *accessToken)
 }
+
+func (r *Resolver) MicrosoftTeamsBotEndpoint(w http.ResponseWriter, req *http.Request) {
+	fmt.Println("Running MSTEAMS Bot")
+	ctx := req.Context()
+
+	bot, err := microsoft_teams.NewMicrosoftTeamsBot("common")
+
+	if err != nil {
+		log.WithContext(ctx).Error(e.Wrap(err, "error making ms bot adapter"))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	act, err := bot.ParseRequest(ctx, req)
```

Comment:
This effectively "validates" and authenticates the request using our client id and secret.
It also extracts the payload into a friendly form that we are able to interact with.